igvc_navigation/src/path_planner/GraphSearch.hpp

```diff
@@ -1,17 +1,46 @@
 #ifndef GRAPHSEARCH_HPP_INCLUDED
 #define GRAPHSEARCH_HPP_INCLUDED
 
+/**
+ * Generalized implementation of multiple Search Algorithms using the abstraction set up in the
+ * problem section of the code
+ */
+
 #include <algorithm>
 #include <iostream>
 #include <list>
 #include <queue>
 #include <set>
 #include <stack>
+#include <map> // TODO move to unordered map
 
 #include "SearchProblem.hpp"
 
+// TODO this is kind of an issue
 using namespace std;
 
+template <class StateType, class ActionType>
+class StateComparator {
```

Comment:
This stuff here is unrelated to the Smooth Control stuff right? I think it belongs in another PR so we can keep the PRs focused? @JasonGibson274

Reply:
ideally yes, but it is probably fine as is. this all goes into devel and that can be kind of chaotic.